p2p/src/lib.rs

```diff
@@ -164,12 +167,29 @@ pub async fn make_p2p<S: PeerDbStorage + 'static>(
     time_getter: TimeGetter,
     peerdb_storage: S,
 ) -> Result<Box<dyn P2pInterface>> {
-    let stream_adapter = NoiseEncryptionAdapter::gen_new();
-    let base_transport = net::default_backend::transport::TcpTransportSocket::new();
-    let transport = NoiseTcpTransport::new(stream_adapter, base_transport);
+    let transport = make_p2p_transport();
 
-    let p2p = P2p::<DefaultNetworkingService<NoiseTcpTransport>>::new(
+    let bind_addresses = if !p2p_config.bind_addresses.is_empty() {
```

Comment:
Perhaps this logic should be moved to a separate function.

Also I keep thinking whether we want BindAddresses to be a trait with a Default impl coming from the application... I don't know. But seems like it's "its own thing".

Reply:
Sure, updated.